manifests/cli.pp

```diff
@@ -21,21 +21,49 @@
   Class['jenkins::service'] ->
     Class['jenkins::cli'] ->
       Anchor['jenkins::end']
-
   $jar = "${jenkins::libdir}/jenkins-cli.jar"
-  $extract_jar = "jar -xf ${jenkins::libdir}/jenkins.war WEB-INF/jenkins-cli.jar"
-  $move_jar = "mv WEB-INF/jenkins-cli.jar ${jar}"
-  $remove_dir = 'rm -rf WEB-INF'
-
-  exec { 'jenkins-cli' :
-    command => "${extract_jar} && ${move_jar} && ${remove_dir}",
-    path    => ['/bin', '/usr/bin'],
-    cwd     => '/tmp',
-    creates => $jar,
-    require => Service['jenkins'],
+  case $::osfamily {
+    'Windows': {
+      $extract_jar = "start-sleep 30; \$j=[Environment]::GetEnvironmentVariable('JAVA_HOME','machine'); & \"\$j/bin/jar\" -xf \"${jenkins::libdir}/jenkins.war\" WEB-INF/jenkins-cli.jar"
+      $move_jar    = "move-item WEB-INF/jenkins-cli.jar '${jar}'"
+      $remove_dir  = 'remove-item -recurse -force WEB-INF'
+      $and_var	   = ';'
+      $path	   = ''
+      $cwd	   = $jenkins::libdir
+      $sleep_cmd  = 'start-sleep -seconds 10'
+    }
+    default: {
+      $extract_jar = "jar -xf ${jenkins::libdir}/jenkins.war WEB-INF/jenkins-cli.jar"
+      $move_jar    = "mv WEB-INF/jenkins-cli.jar ${jar}"
+      $remove_dir  = 'rm -rf WEB-INF'
+      $and_var     = '&&'
+      $path	   = "['/bin', '/usr/bin']"
+      $cwd	   = '/tmp'
+      $sleep_cmd  = '/bin/sleep 10'
+    }
   }
-
-  file { $jar:
+  
+  if ($::operatingsystem == 'windows')  {
```

Comment:
This if statement is only to set the provider param?  Why not just set the param from a variable?